### PR TITLE
[SYCLomatic] Fix bug issues met in the migration of CudaSift's cmake script

### PR DESCRIPTION
**Pull request by tomflinda**
1. Fix issue that "find_package(OpenCV REQUIRED)" is migrated to "find_package(IntelSYCL REQUIRED)", which results in the compile issue that ${OpenCV_LIBS} undefined.
2. Fix issue that cmake_minimum_required(VERSION 3.10) is not migrated.
3. Fix issue that mainSift.cpp which has cuda syntax, the name extension is not changed to mainSift.cpp.dp.cpp
4. Emit processing msg for each cmake script

### CODE REVIEW

File: clang/lib/DPCT/DPCT.cpp [hunk 1]
```diff
@@ -536,6 +537,23 @@ void parseFormatStyle() {
...
+
+  std::string YamlFilePath = appendPath(OutRoot.getCanonicalPath().str(),
+                                        DpctGlobalInfo::getYamlFileName());
+  auto PreTU = std::make_shared<clang::tooling::TranslationUnitReplacements>();
+  if (llvm::sys::fs::exists(YamlFilePath)) {
+    if (loadFromYaml(YamlFilePath, *PreTU) == 0) {
+
+    } else {
+      llvm::errs() << getLoadYamlFailWarning(YamlFilePath);
+    }
```
Comment (zhiweij1) [inline clang/lib/DPCT/DPCT.cpp]:
```suggestion
    if (loadFromYaml(YamlFilePath, *PreTU)) {
      llvm::errs() << getLoadYamlFailWarning(YamlFilePath);
    }
```

File: clang/lib/DPCT/DPCT.cpp [hunk 2]
```diff
@@ -536,6 +537,23 @@ void parseFormatStyle() {
   DpctGlobalInfo::setCodeFormatStyle(Style);
 }
 
+static void loadMainSrcFileInfo(clang::tooling::UnifiedPath OutRoot) {
+
```
Comment (zhiweij1) [inline clang/lib/DPCT/DPCT.cpp]:
```suggestion
```

File: clang/lib/DPCT/MigrateCmakeScript.cpp [hunk 3]
```diff
@@ -512,9 +515,11 @@ void applyImplicitMigrationRule(std::string &Input,
   size_t Size = Input.size();
   size_t Index = 0;
   while (Index < Size) {
-
+    Index = skipWhileSpaces(Input, Index);
```
Comment (zhiweij1) [inline clang/lib/DPCT/MigrateCmakeScript.cpp]: WhileSpace or WhiteSpace ?

File: clang/lib/DPCT/PatternRewriter.cpp [hunk 4]
```diff
@@ -388,6 +393,33 @@ static bool isIdentifiedChar(char Char) {
   return false;
 }
 
+static void
+updateExtentionName(const std::string &Input, size_t Next,
+                    std::unordered_map<std::string, std::string> &Bindings) {
+  if (Input.compare(Next, strlen(".cpp"), ".cpp") == 0) {
```
Comment (zhiweij1) [inline clang/lib/DPCT/PatternRewriter.cpp:401]: How about using llvm API like path::extension?

File: clang/lib/DPCT/DPCT.cpp (hunk 1 above)
Comment (tomflinda) [reply clang/lib/DPCT/DPCT.cpp]: Will update it in the updated PR.

File: clang/lib/DPCT/DPCT.cpp (hunk 2 above)
Comment (tomflinda) [reply clang/lib/DPCT/DPCT.cpp]: Will update it in the updated PR.

File: clang/lib/DPCT/MigrateCmakeScript.cpp (hunk 3 above)
Comment (tomflinda) [reply clang/lib/DPCT/MigrateCmakeScript.cpp]: Will update it in the updated PR.

File: clang/lib/DPCT/PatternRewriter.cpp (hunk 4 above)
Comment (tomflinda) [reply clang/lib/DPCT/PatternRewriter.cpp:401]: Input is not  a file path, it is a string buffer which include file name, so path::extension does not work.

**Review by zhiweij1 — APPROVED** [review]

**Review by AndyCHHuang — APPROVED** [review]
LGTM

File: clang/lib/DPCT/PatternRewriter.cpp [hunk 5]
```diff
@@ -388,6 +393,33 @@ static bool isIdentifiedChar(char Char) {
...
+    std::string SyclFileName;
+    rewriteFileName(SyclFileName, FileName);
+    bool HasCudaSyntax = false;
+    for (const auto &File : MainSrcFilesHasCudaSyntex) {
+      if (File.find(FileName) != std::string::npos) {
+        HasCudaSyntax = true;
+      }
+    }
+    if (HasCudaSyntax)
+      Bindings["rewrite_extention_name"] = "cpp.dp.cpp";
```
Comment (zhimingwang36) [inline clang/lib/DPCT/PatternRewriter.cpp]: Please fix the hardcode name in following PR in L415.

File: clang/tools/dpct/DpctOptRules/cmake_script_migration_rule.yaml [hunk 6]
```diff
@@ -111,18 +99,32 @@
...
   Priority: Fallback
   MatchMode: Partial
-  CmakeSyntax: cuda_files_in_user_defined_function
+  CmakeSyntax: cuda_files_in_user_defined_function_cpp
+  In: ${func_name}(${value})
+  Out: ${func_name}(${value})
+  Subrules:
+    value:
+      MatchMode: Full
+      In: ${arg}.cpp
```
Comment (zhimingwang36) [inline clang/tools/dpct/DpctOptRules/cmake_script_migration_rule.yaml:112]:
```suggestion
      In: ${arg}.cpp | ${arg}.cxx | ${arg}.cu | ${arg}.C
```

File: clang/lib/DPCT/PatternRewriter.cpp (hunk 4 above)
Comment (zhimingwang36) [inline clang/lib/DPCT/PatternRewriter.cpp:401]: what if .cxx  .C , *c?

File: clang/lib/DPCT/PatternRewriter.cpp (hunk 5 above)
Comment (tomflinda) [reply clang/lib/DPCT/PatternRewriter.cpp]: Will update it.

File: clang/tools/dpct/DpctOptRules/cmake_script_migration_rule.yaml (hunk 6 above)
Comment (tomflinda) [reply clang/tools/dpct/DpctOptRules/cmake_script_migration_rule.yaml:112]: Will implement this feature in a separate PR.

File: clang/lib/DPCT/PatternRewriter.cpp (hunk 4 above)
Comment (tomflinda) [reply clang/lib/DPCT/PatternRewriter.cpp:401]: Will implement this feature in a separate PR.

**Pull request merged**